File: LEETCODE/Medium/Pascal's Triangle/pascalTriangle.py
"""
Problem:
    Given an integer numRows, return the first numRows of Pascal's triangle.
    In Pascal's triangle, each number is the sum of the two numbers directly above it.
"""

from typing import List

def pascal_triangle(numRows: int) -> List[int]:
    """
    :param numRows: int
    :return: List[int]
    """
    triangle = []
    for line in range(1, numRows + 1):
        c = 1
        term = []
        for i in range(1, line + 1):
            term.append(c)
            c = c * (line - i) // i
        triangle.append(term)
    
    return triangle


if __name__ == '__main__':
    print(pascal_triangle(8))

File: LEETCODE/Medium/Pascal's Triangle II/pascalTriangle.java
import java.util.ArrayList;
import java.util.List;

/*
    Problem:
        Given an integer rowIndex, return the rowIndexth (0-indexed) row of the Pascal's triangle.

        In Pascal's triangle, each number is the sum of the two numbers directly above it
*/

public class pascalTriangle {
    public static void main(String[] args) {
        System.out.println("7th row from pascal triangle: " + getRow(7).toString());
    }

    public static List<Integer> getRow(int rowIndex) { 
        long c = 1l;
        List<Integer> row = new ArrayList<>();
        for (int i = 1; i <= rowIndex + 1; i++) {
            row.add((int)c);
            c = c * (rowIndex - i + 1) / i;
        }
        
        return row;
    }
}